### PR TITLE
Refx/property improvements

### PR DESCRIPTION
**Pull request by FigBug**
Made more properties settable
Added named properties to the list
Added colour picker for juce colours

### CODE REVIEW

**Comment by sudara** [comment]
Thanks, I'll take a look! 

> Feels backwards right now.

Fair enough, it's an artifact of how it was first built (everything in one col). I was going to start off by putting the right column into it's own wrapper component (see: the unused tree_component) but got distracted... maybe @dikadk remembers if there was another reason for this layout, I could be forgetting.

**Comment by sudara** [comment]
Two questions for you:

* Do you have colors throughout the component hierarchy or do they all tend to be at the top level? 
* Are you using the functionality you added where you can pass in a PropertiesFile* to melatonin::inspector's constructor or was it just to offer the option? If you aren't using, I might simplify (need to pass the settings to components or use a `SharedResourcePointer`)

File: melatonin/components/component_tree_view_item.h [hunk 1]
```diff
@@ -48,6 +48,43 @@ namespace melatonin
...
+                    67,0,0,112,67,98,0,0,224,67,51,51,103,67,154,153,227,67,0,0,96,67,0,0,232,67,0,0,96,67,108,0,0,248,67,0,0,96,67,98,102,102,252,67,0,0,96,67,0,0,0,68,51,51,103,67,0,0,0,68,0,0,112,67,108,0,0,0,68,0,0,136,67,98,0,0,0,68,102,102,140,67,102,102,252,67,0,
+                    0,144,67,0,0,248,67,0,0,144,67,108,0,0,232,67,0,0,144,67,98,154,153,227,67,0,0,144,67,0,0,224,67,102,102,140,67,0,0,224,67,0,0,136,67,108,0,0,224,67,0,0,112,67,99,109,0,0,232,67,0,0,160,67,108,0,0,248,67,0,0,160,67,98,102,102,252,67,0,0,160,67,0,0,0,
+                    68,154,153,163,67,0,0,0,68,0,0,168,67,108,0,0,0,68,0,0,184,67,98,0,0,0,68,102,102,188,67,102,102,252,67,0,0,192,67,0,0,248,67,0,0,192,67,108,0,0,232,67,0,0,192,67,98,154,153,227,67,0,0,192,67,0,0,224,67,102,102,188,67,0,0,224,67,0,0,184,67,108,0,0,224,
+                    67,0,0,168,67,98,0,0,224,67,154,153,163,67,154,153,227,67,0,0,160,67,0,0,232,67,0,0,160,67,99,101,0,0 };
+
```
Comment (sudara) [inline melatonin/components/component_tree_view_item.h:79]: Loving this idea but is it reliable for you? When I click back to the inspector, keyboard focus is lost, making it feel  glitchy. Wondering if we can pepper around `setMouseClickGrabsKeyboardFocus(false)` to solve (or have the icon also show "last focus").

Also, I'm missing seeing wantsKeyboardFocus in the hierarchy, so I guess I'd put in an less bright icon for that and move the icons to the end of the label unless you object.

**Comment by FigBug** [comment]
> Also, I'm missing seeing wantsKeyboardFocus in the hierarchy, so I guess I'd put in an less bright icon for that and move the icons to the end of the label unless you object.

Originally I had the icon at the end, I thought it looked funny. But it doesn't really matter to me.

**Comment by FigBug** [comment]
> Do you have colors throughout the component hierarchy or do they all tend to be at the top level?

throughout

> Are you using the functionality you added where you can pass in a PropertiesFile* to melatonin::inspector's constructor or was it just to offer the option? If you aren't using, I might simplify (need to pass the settings to components or use a SharedResourcePointer)

I'm passing nullptr, don't use the juce propertiesfile in my app

**Comment by sudara** [comment]
Closing, as these commits are included in #48

**Pull request closed without merging**